### ISSUE TITLE
ReactCSSTransitionGroup throws error when using a timeout and the element has been removed

### ISSUE THREAD

**Issue by andersekdahl** [facebook/react#5499]
Using React 0.14.2.

With this render method:
```
  render() {
    const className = this.props.className;
    if (this.props.isVisible) {
      return (
        <ReactCSSTransitionGroup
          transitionName={{
            enter: 'enter',
            leave: 'leave',
            appear: 'appear',
          }}
          transitionAppear
          transitionAppearTimeout={200}
          transitionEnterTimeout={0}
          transitionLeaveTimeout={0}
        >
          <Spinner key="spinner" className={className} />
        </ReactCSSTransitionGroup>
      );
    } else {
      return null;
    }
  }
```

An exception is thrown if `this.props.isVisible` becomes `false` before the `transitionAppearTimeout` has has elapsed.

The exception occurs in `ReactCSSTransitionGroup._handleDoneAppearing` where it tries to find the component, but `this.refs` is empty. I'm guessing that `ReactCSSTransitionGroup._handleDoneAppearing` should just exit early if the components are gone.

**Comment by johanneslumpe** [comment]
I am too encountering a similar issue, but not in `_handleDoneAppearing` but `_handleDoneEntering`:

```javascript
    var component = this.refs[key];
    if (component.componentDidEnter) {
      component.componentDidEnter();
    }
```

The error I'm getting: 
`Uncaught TypeError: Cannot read property 'componentDidEnter' of undefined`

A simple solution would be to check the existence and exit, as mentioned by the OP.

**Comment by schovi** [comment]
@andersekdahl @jimfb We tried to debug this problem and find that it is already fixed in 0.14.3 with this pull request https://github.com/facebook/react/pull/5391 It is not relevant anymore.

**Comment by andersekdahl** [comment]
Yup, seems like it, thanks!
